#!/usr/bin/env python
PACKAGE = "legged_controllers"
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("kp_position",    double_t,    0,  "Kp_position",    100,    0,   300)
gen.add("kd_position",    double_t,    1,  "Kd_position",    3,    0,   100)
gen.add("kp_big_stance",    double_t,    2,  "kp_big_stance",    40,    0,   100)
gen.add("kp_big_swing",    double_t,    3,  "kp_big_swing",    30,    0,   100)
gen.add("kd_big",    double_t,    4,  "kd_big",    2,    0,   20)

gen.add("kp_small_stance",    double_t,    5,  "kp_small_stance",    30,    0,   100)
gen.add("kp_small_swing",    double_t,    6,  "kp_small_swing",    20,    0,   100)
gen.add("kd_small",    double_t,    7,  "kd_small",    2,    0,   20)

gen.add("kd_feet",    double_t,    10,  "kd_feet",    0.01,    0,   20)

gen.add("kp_stance1",    double_t,    11,  "kp_stance1",    500,    0,   1000)
gen.add("kp_stance2",    double_t,    12,  "kp_stance2",    110,    0,   1000)
gen.add("kp_stance3",    double_t,    13,  "kp_stance3",    280,    0,   1000)
gen.add("kp_stance4",    double_t,    14,  "kp_stance4",    300,    0,   1000)
gen.add("kp_stance5",    double_t,    15,  "kp_stance5",    680,    0,   1000)

gen.add("kd_stance1",    double_t,    16,  "kd_stance1",    5.0,    0,   20)
gen.add("kd_stance2",    double_t,    17,  "kd_stance2",    1.6,    0,   20)
gen.add("kd_stance3",    double_t,    18,  "kd_stance3",    2.0,    0,   20)
gen.add("kd_stance4",    double_t,    19,  "kd_stance4",    1.5,    0,   20)
gen.add("kd_stance5",    double_t,    20,  "kd_stance5",    1.7,    0,   20)
gen.add("kd_stance6",    double_t,    21,  "kd_stance6",    0.5,    0,   20)

gen.add("kp_mpc1",    double_t,    22,  "kp_mpc1",    250,    0,   1000)
gen.add("kp_mpc2",    double_t,    23,  "kp_mpc2",    100,    0,   1000)
gen.add("kp_mpc3",    double_t,    24,  "kp_mpc3",    300,    0,   1000)
gen.add("kp_mpc4",    double_t,    25,  "kp_mpc4",    200,    0,   1000)

gen.add("kd_mpc1",    double_t,    26,  "kd_mpc1",    3.01,    0,   20)
gen.add("kd_mpc2",    double_t,    27,  "kd_mpc2",    2.01,    0,   20)
gen.add("kd_mpc3",    double_t,    28,  "kd_mpc3",    1.01,    0,   20)

exit(gen.generate(PACKAGE, "legged_controllers", "Tutorials")) 